src/build/release.rs

```diff
@@ -26,50 +29,83 @@ pub fn release(p: &crate::project::Project, version: &String) -> Result<(), Erro
         }
     }
 
-    // Generate key
     if !Path::new("releases/keys").exists() {
         fs::create_dir("releases/keys")?;
     }
+
+    let mut key;
```

Comment:
I'm surprised the compiler allowed this, variables should always be given a value when they are defined. This should just be down on line 54.

Reply:
It's just a lazy way of allowing the two cases of `key` to store their result in a variable that the rest of the programme can read. If I put it down further, it'd go out of scope.

I guess the "right" way would be to do `let key = give_me_the_key(reuse_private_key: bool, ...)` and put the if below into it?

I don't really see the issue with how it is currently.

Reply:
In rust `if` statements can return values. The if should return a value rather than using an empty value.

```rust
let x: int;
if something {
	case1();
	x = 1;
} else {
	case2();
	x = 2;
}
```
should be
```rust
let x = if something {
	case1();
	1
} else {
	case2();
	2
};
```

Reply:
Oh, that's magic. Thanks.